eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -123,86 +51,9 @@ jobs:
-
-      - script: |
-          npm ci
-        workingDirectory: $(System.DefaultWorkingDirectory)/eng/tools/analyze-deps
-        displayName: "Install tool dependencies"
-
-      - script: |
-          node index.js --verbose --dump "$(Build.ArtifactStagingDirectory)/data.js" --out "$(Build.ArtifactStagingDirectory)/dependencies.html"
-        workingDirectory: $(System.DefaultWorkingDirectory)/eng/tools/analyze-deps
-        displayName: "Analyze library dependencies"
-
-      - script: |
-          node common/scripts/install-run-rush.js install
-        displayName: "Install library dependencies"
-
-      - pwsh: |
-          node common/scripts/install-run-rush.js build -t @azure/eslint-plugin-azure-sdk
-          node eng/tools/rush-runner.js lint "${{parameters.ServiceDirectory}}"
-        displayName: "Build ESLint Plugin and Lint Libraries"
-
-      - script: |
-          node eng/tools/rush-runner.js audit "${{parameters.ServiceDirectory}}"
-        condition: and(succeeded(), eq(variables['RunNpmAudit'], 'true'))
-        displayName: "Audit libraries"
-
-      - ${{ each artifact in parameters.Artifacts }}:
-        - template: /eng/common/pipelines/templates/steps/verify-changelog.yml
-          parameters:
-            PackageName: ${{artifact.name}}
-            ServiceName: ${{parameters.ServiceDirectory}}
-            ForRelease: false
-
-      - task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
-        # ComponentGovernance is currently unable to run on pull requests of public projects. Running on non-PR
-        # builds should be sufficient.
-        condition: and(succeededOrFailed(), ne(variables['Build.Reason'], 'PullRequest'))
-        displayName: "Component Detection"
-
-      # Unlink node_modules folders to significantly improve performance of subsequent tasks
-      # which need to walk the directory tree (and are hardcoded to follow symlinks).
-      - script: |
-          node eng/tools/rush-runner.js unlink
-        displayName: "Unlink dependencies"
-
-      # It's important for performance to pass "sdk" as "sourceFolder" rather than as a prefix in "contents".
-      # The task first enumerates all files under "sourceFolder", then matches them against the "contents" pattern.
-      - task: CopyFiles@2
-        inputs:
-          sourceFolder: sdk
-          contents: "**/**/*lintReport.html"
-          targetFolder: $(Build.ArtifactStagingDirectory)
-          flattenFolders: true
-        displayName: "Copy lint reports"
-
-      - task: PublishPipelineArtifact@1
-        condition: succeededOrFailed()
-        displayName: "Publish Report Artifacts"
-        inputs:
-          artifactName: reports
-          path: $(Build.ArtifactStagingDirectory)
+          ServiceDirectory: ${{parameters.ServiceDirectory}}
 
   # Only run tests if the matrix has entries
   - ${{ if ne(parameters.RunUnitTests, false) }}:
```

Comment:
Why are you not also factoring out the tests?

Reply:
I was planning on doing this in next PR but actually I will make this change as well in this PR.